### PR TITLE
`gw-cache-buster.php`: Added new `post_render_script` override for GravityForms change.

### PR DESCRIPTION
**Pull request by veryspry**
## Context

💬 Slack: https://gravitywiz.slack.com/archives/C03T0CNL8AV/p1709131136355889 

## Summary

Gravity Forms rolled out this change which we now need to account for here.

https://github.com/gravityforms/gravityforms/pull/2762

### CODE REVIEW

File: gravity-forms/gw-cache-buster.php [hunk 1]
```diff
@@ -205,6 +209,14 @@ public function suppress_default_post_render_event( $form_string, $form, $curren
 			$form_string = str_replace( $search, '', $form_string );
 		}
 
+		if ( is_callable( 'GFFormDisplay::post_render_script' ) ) {
+			$post_render_script = GFFormDisplay::post_render_script( $form['id'], $current_page );
+			// $post_render_script    = GFCommon::get_inline_script_tag( $post_render_script );
+			$post_render_script = preg_quote( $post_render_script, '/' ); // Escape special characters
+			$pattern            = '/gform\.initializeOnLoaded\(\s*function\(\)\s*\{\s*(' . $post_render_script . ')\s*\}\s*\);/';
+			$form_string        = preg_replace( $pattern, '', $form_string );
```
Comment (veryspry) [inline gravity-forms/gw-cache-buster.php:217]: This does leave an empty `<script></script>` in `$form_string`. I will address that when we confirm that this fix is going to work but I don't want to waste time on that if the approach needs changed.

File: gravity-forms/gw-cache-buster.php [hunk 2]
```diff
@@ -177,8 +177,13 @@ public function shortcode( $markup, $attributes, $content ) {
...
 						// Form has been rendered. Trigger post render to initialize scripts.
-						jQuery( document ).trigger( 'gform_post_render', [ formId, 1 ] );
-						gform.utils.trigger({ event: 'gform/postRender', native: false, data: { formId: formId, currentPage: 1 } });
+						<?php
+							echo 'gform.initializeOnLoaded( function() {' . GFFormDisplay::post_render_script(
+								$form_id,
+								GFFormDisplay::get_current_page( $form_id )
+								) .
+								'} );';
+						?>
```
Comment (spivurno) [inline gravity-forms/gw-cache-buster.php:186]:
```suggestion
						<?php
							$post_render_script = GFFormDisplay::post_render_script( $form_id, GFFormDisplay::get_current_page( $form_id ) );
							print_f( 'gform.initializeOnLoaded( function() { %s } )', $post_render_script );
						?>
```

Dude, please don't hate me for making a formatting suggestion but I find `print_f` so much easier to read, particularly when there's a multi-line function in play. Thoughts on something like this?

File: gravity-forms/gw-cache-buster.php [hunk 3]
```diff
@@ -205,6 +209,13 @@ public function suppress_default_post_render_event( $form_string, $form, $curren
 			$form_string = str_replace( $search, '', $form_string );
 		}
 
+		if ( is_callable( 'GFFormDisplay::post_render_script' ) ) {
```
Comment (spivurno) [inline gravity-forms/gw-cache-buster.php:212]: Didn't know you could use `is_callable` on a string like this. Nice!

File: gravity-forms/gw-cache-buster.php (hunk 1 above)
Comment (spivurno) [reply gravity-forms/gw-cache-buster.php:216]: This works for me!

**Review by spivurno — APPROVED** [review]
Approved with on teeny tiny suggestion.

**Comment by veryspry** [comment]
@spivurno thanks for the review! I will get your suggestions addressed and this merged by tomorrow 🙏

File: gravity-forms/gw-cache-buster.php (hunk 2 above)
Comment (veryspry) [reply gravity-forms/gw-cache-buster.php]: Haha no sweat 😉  Change made

File: gravity-forms/gw-cache-buster.php (hunk 1 above)
Comment (veryspry) [reply gravity-forms/gw-cache-buster.php:217]: Resolved in 824c17801a6b0736efbb121d2ba634dc09fb2610

**Comment by veryspry** [comment]
The linting error is unrelated to these changes.

**Pull request merged**